usaspending_api/recipient/delta_models/recipient_profile.py

```diff
@@ -1,26 +1,434 @@
+        unused = false;""",
+    # --------------------------------------------------------------------------------
+    # -- Step 6, Populate parent recipient list in children
+    # --------------------------------------------------------------------------------
+    """WITH all_recipients AS (
+        SELECT
+            uei,
+            COLLECT_SET(DISTINCT parent_uei) AS parent_uei_list
+        FROM
+            temporary_recipients_from_transactions_view
+        WHERE
+            uei IS NOT NULL AND
+            parent_uei IS NOT NULL
+        GROUP BY uei
+    )
+    MERGE INTO temporary_restock_recipient_profile AS rpv
+    USING all_recipients AS ar
+    ON rpv.uei = ar.uei AND
+        rpv.recipient_level = 'C'
+    WHEN MATCHED THEN UPDATE SET
+        recipient_affiliations = ar.parent_uei_list,
+        unused = false;""",
+    # --------------------------------------------------------------------------------
+    # -- Step 7, Mark recipient profile rows older than 12 months  as valid
+    # --------------------------------------------------------------------------------
+    """WITH grouped_by_old_recipients AS (
+        SELECT
+            recipient_hash,
+            recipient_level
+        FROM
+            temporary_recipients_from_transactions_view AS trft
+        GROUP BY recipient_hash, recipient_level
+    )
+
+    MERGE INTO temporary_restock_recipient_profile AS rpv
+    USING grouped_by_old_recipients AS gbc
+    ON gbc.recipient_hash = rpv.recipient_hash AND
+        gbc.recipient_level = rpv.recipient_level AND
+        rpv.unused = true
+    WHEN MATCHED THEN UPDATE SET
+        unused = false;""",
+    # --------------------------------------------------------------------------------
+    # -- Step 8, Mark Parent recipient profile rows older than 12 months  as valid
+    # --------------------------------------------------------------------------------
+    """WITH grouped_by_parent_old AS (
+        SELECT
+            parent_uei
+        FROM
+            temporary_recipients_from_transactions_view AS trft
+        WHERE
+            parent_uei IS NOT NULL
+        GROUP BY parent_uei
+    )
+    MERGE INTO temporary_restock_recipient_profile AS rpv
+    USING grouped_by_parent_old AS gbp
+    ON rpv.uei = gbp.parent_uei AND
+        rpv.recipient_level = 'P' AND
+        rpv.unused = true
+    WHEN MATCHED THEN UPDATE SET
+        unused = false;""",
+    # --------------------------------------------------------------------------------
+    # -- Step 9, Finalize new table
+    # --------------------------------------------------------------------------------
+    """DELETE FROM temporary_restock_recipient_profile WHERE unused = true;""",
+    # --------------------------------------------------------------------------------
+    # -- Step 10, Drop unnecessary relations and standup new table as final
+    # --------------------------------------------------------------------------------
```

Comment:
What's the purpose of populating a temporary table, `temporary_restock_recipient_profile`, then performing a series of merges to make `recipient_profile` match? 
I believe this was done in Postgres to prevent a live table from being updated by a series of steps (1-9) that could fail, but because this DBR ETL is updating a table that will later be nuke/paved, would it make sense to clear `recipient_profile` at the start of the script and update directly?

Reply:
yeah that is fair, i'll see what i can do